Unexpected keyword argument 'request' in streaming method BidiReadObject

#### Affected API
`//google/storage/v2/storage_py_gapic_test`


#### Failure Log
[build log link](https://fusion2.corp.google.com/ci/kokoro/prod:actools%2Fgoogleapis%2Fstable%2Finternal%2Fpresubmit/activity/ffe177b8-8eef-4b02-937e-2b1ec47152c4/log)
```
=================================== FAILURES ===================================
___________ test_bidi_read_object_routing_parameters_request_1_grpc ____________

    def test_bidi_read_object_routing_parameters_request_1_grpc():
        client = StorageClient(
            credentials=ga_credentials.AnonymousCredentials(),
            transport="grpc",
        )
    
        # Mock the actual call, and fake the request.
        with mock.patch.object(
                type(client.transport.bidi_read_object),
                '__call__') as call:
            call.return_value = iter([storage.BidiReadObjectResponse()])
>           client.bidi_read_object(request={"read_object_spec": {"bucket": "sample1"}})
E           TypeError: StorageClient.bidi_read_object() got an unexpected keyword argument 'request'

google/storage/v2/[storage_py_gapic_srcjar.py/tests/unit/gapic/storage_v2/test_storage.py:8188](https://www.google.com/url?q=http://storage_py_gapic_srcjar.py/tests/unit/gapic/storage_v2/test_storage.py:8188&sa=D): TypeError
_______ test_bidi_read_object_routing_parameters_request_1_grpc_asyncio ________

    @pytest.mark.asyncio
    async def test_bidi_read_object_routing_parameters_request_1_grpc_asyncio():
        client = StorageAsyncClient(
            credentials=async_anonymous_credentials(),
            transport="grpc_asyncio",
        )
    
        # Mock the actual call, and fake the request.
        with mock.patch.object(
                type(client.transport.bidi_read_object),
                '__call__') as call:
            # Designate an appropriate return value for the call.
            call.return_value = mock.Mock(aio.UnaryStreamCall, autospec=True)
            call.return_value.read = mock.AsyncMock(side_effect=[storage.BidiReadObjectResponse()])
>           await client.bidi_read_object(request={"read_object_spec": {"bucket": "sample1"}})
E           TypeError: StorageAsyncClient.bidi_read_object() got an unexpected keyword argument 'request'

google/storage/v2/[storage_py_gapic_srcjar.py/tests/unit/gapic/storage_v2/test_storage.py:9657](https://www.google.com/url?q=http://storage_py_gapic_srcjar.py/tests/unit/gapic/storage_v2/test_storage.py:9657&sa=D): TypeError
```

--- a/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2
+++ b/gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2
@@ -2173,7 +2173,8 @@ def test_initialize_client_w_{{transport_name}}():
 
 {% macro routing_parameter_test(service, api, transport, is_async) %}
 {% for method in service.methods.values() %}{# method #}
-{% if method.explicit_routing %}
+{# See existing proposal b/330610501 to add support for explicit routing in BIDI/client side streaming #}
+{% if method.explicit_routing and not method.client_streaming %}
 {# Any value that is part of the HTTP/1.1 URI should be sent as #}
 {# a field header. Set these to a non-empty value. #}
 {% for routing_param in method.routing_rule.routing_parameters %}

